From 28607b7eb80cb5e1ba8a0a8c5f648c4ba16721fc Mon Sep 17 00:00:00 2001
From: Mark Enriquez <enriquezmark36@gmail.com>
Date: Fri, 14 Dec 2018 13:22:51 +0800
Subject: Revert all destructive changes from the patchset

So far, the patchset should make it work for kanas devices but it would
probably affect other phones like by causing some compilation error
or segfaulting libraries, particularly in the audio/video system.

Hopefully, this commit reverts most, if not all, changes that affects how
codepaths are accessed for other phones.

Signed-off-by: Mark Enriquez <enriquezmark36@gmail.com>
---
 audio/sc8830/eng_audio.h                             | 1 +
 gralloc/Android.mk                                   | 6 +-----
 libcamera/Android.mk                                 | 4 ++--
 libmemoryheapion/Android.mk                          | 8 +++++---
 omx_components/video/avc_sprd/Android.mk             | 7 ++++---
 omx_components/video/colorformat_switcher/Android.mk | 9 ++++++++-
 omx_components/video/vpx_sprd/Android.mk             | 9 ++++++++-
 7 files changed, 29 insertions(+), 15 deletions(-)

diff --git a/audio/sc8830/eng_audio.h b/audio/sc8830/eng_audio.h
index a060f9b..8d48b23 100644
--- a/audio/sc8830/eng_audio.h
+++ b/audio/sc8830/eng_audio.h
@@ -1,6 +1,7 @@
 #ifndef __ENG_AUDIO_H__
 #define __ENG_AUDIO_H__
 
+#include <system/audio.h>
 
 #define NAME_LEN_MAX                            16
 #define EQ_BAND_MAX				8
diff --git a/gralloc/Android.mk b/gralloc/Android.mk
index 361b3f3..c0c55e2 100644
--- a/gralloc/Android.mk
+++ b/gralloc/Android.mk
@@ -21,12 +21,8 @@ supported_boards := \
 	sc8830 \
 	scx15 \
 
-ifeq ($(SOC_SCX30G_V2),true)
-include $(call all-named-subdir-makefiles,scx30g_v2)
-else ifneq (,$(filter $(supported_boards),$(TARGET_BOARD_PLATFORM)))
 ifeq ($(SOC_SCX35),true)
 include $(call all-named-subdir-makefiles,scx15)
-else
+else ifneq (,$(filter $(supported_boards),$(TARGET_BOARD_PLATFORM)))
 include $(call all-named-subdir-makefiles,$(TARGET_BOARD_PLATFORM))
 endif
-endif
diff --git a/libcamera/Android.mk b/libcamera/Android.mk
index 76b5687..67795b7 100644
--- a/libcamera/Android.mk
+++ b/libcamera/Android.mk
@@ -30,5 +30,5 @@ include $(call all-named-subdir-makefiles,scx15)
 endif
 
 ifeq ($(SOC_SCX30G_V2),true)
-include $(call all-makefiles-under,sc8830)
-endif
\ No newline at end of file
+include $(call all-named-subdir-makefiles,sc8830)
+endif
diff --git a/libmemoryheapion/Android.mk b/libmemoryheapion/Android.mk
index 862b1dd..862580e 100644
--- a/libmemoryheapion/Android.mk
+++ b/libmemoryheapion/Android.mk
@@ -19,8 +19,10 @@ supported_boards := \
 	sc8830 \
 	scx15 \
 
-ifeq ($(SOC_SCX30G_V2),true)
-include $(call all-named-subdir-makefiles,sc8830)
-else ifneq (,$(filter $(supported_boards),$(TARGET_BOARD_PLATFORM)))
+ifneq (,$(filter $(supported_boards),$(TARGET_BOARD_PLATFORM)))
+ifeq ($(SOC_SCX35),true)
 include $(call all-named-subdir-makefiles,scx15)
+else
+include $(call all-named-subdir-makefiles,$(TARGET_BOARD_PLATFORM))
+endif
 endif
diff --git a/omx_components/video/avc_sprd/Android.mk b/omx_components/video/avc_sprd/Android.mk
index 7eb2f71..657f347 100644
--- a/omx_components/video/avc_sprd/Android.mk
+++ b/omx_components/video/avc_sprd/Android.mk
@@ -22,12 +22,13 @@ ifneq (,$(filter scx15 sc8830,$(TARGET_BOARD_PLATFORM)))
 include_makefiles := \
 	$(LOCAL_PATH)/thumbnail/Android.mk \
 
+# TODO Sync sc8830 and scx15 encoder codes
+
 ifeq (sc8830,$(TARGET_BOARD_PLATFORM))
 ifeq ($(SOC_SCX35),true)
 include_makefiles += $(call all-named-subdir-makefiles,scx15 sc8830/dec)
-else
-ifeq ($(SOC_SCX30G_V2),true) # For scx30g_v2 board (sc8830)
-include_makefiles += $(call all-named-subdir-makefiles,scx30g_v2)
+else ifeq ($(SOC_SCX30G_V2),true) # For scx30g2 board (sc8830)
+include_makefiles += $(call all-named-subdir-makefiles,scx30g2)
 else
 include_makefiles += $(call all-named-subdir-makefiles,sc8830)
 endif
diff --git a/omx_components/video/colorformat_switcher/Android.mk b/omx_components/video/colorformat_switcher/Android.mk
index 0c6c9d1..7ce1faa 100644
--- a/omx_components/video/colorformat_switcher/Android.mk
+++ b/omx_components/video/colorformat_switcher/Android.mk
@@ -27,10 +27,17 @@ LOCAL_MODULE := libcolorformat_switcher
 LOCAL_MODULE_TAGS := optional
 
 LOCAL_C_INCLUDES += \
-	$(TOP)/hardware/sprd/gralloc/scx15 \
 	$(TOP)/frameworks/av/include/media/ \
 	$(TOP)/frameworks/native/include/media/openmax
 
+ifeq ($(strip $(SOC_SCX35)),true)
+LOCAL_C_INCLUDES += \
+       $(TOP)/hardware/sprd/gralloc/sc8830
+else
+LOCAL_C_INCLUDES += \
+       $(TOP)/hardware/sprd/gralloc/scx15
+endif
+
 LOCAL_EXPORT_C_INCLUDE_DIRS := $(LOCAL_PATH)
 
 LOCAL_SRC_FILES := colorformat_switcher.cpp
diff --git a/omx_components/video/vpx_sprd/Android.mk b/omx_components/video/vpx_sprd/Android.mk
index 9c784da..a9067b8 100644
--- a/omx_components/video/vpx_sprd/Android.mk
+++ b/omx_components/video/vpx_sprd/Android.mk
@@ -18,5 +18,12 @@
 LOCAL_PATH := $(call my-dir)
 
 ifneq (,$(filter scx15 sc8830,$(TARGET_BOARD_PLATFORM)))
-include $(call all-named-subdir-makefiles,sc8830)
+ifeq ($(SOC_SCX30G_V2),true) # For scx30g2 board (sc8830)
+include_makefiles += $(call all-named-subdir-makefiles,scx30g2)
+else
+include_makefiles += $(call all-named-subdir-makefiles,sc8830)
 endif
+endif
+
+include $(include_makefiles)
+
-- 
2.19.1

